crates/provider/src/native/node.rs

```diff
@@ -1,58 +1,259 @@
 use std::{
     path::{Path, PathBuf},
-    sync::Arc,
+    process::Stdio,
+    sync::{Arc, Weak},
     time::Duration,
 };
 
 use anyhow::anyhow;
 use async_trait::async_trait;
-use nix::{sys::signal::Signal, unistd::Pid};
-use support::{
-    fs::FileSystem,
-    process::{Command, DynProcess, ProcessManager},
+use futures::future::try_join_all;
+use nix::{
+    sys::signal::{kill, Signal},
+    unistd::Pid,
+};
+use support::fs::FileSystem;
+use tokio::{
+    io::{AsyncRead, AsyncReadExt, BufReader},
+    process::{Child, ChildStderr, ChildStdout, Command},
+    sync::{
+        mpsc::{self, Sender},
+        RwLock,
+    },
+    task::JoinHandle,
+    time::sleep,
+    try_join,
 };
-use tokio::{sync::RwLock, task::JoinHandle, time::sleep};
 
-use super::{helpers::create_process_with_log_tasks, namespace::WeakNativeNamespace};
+use super::namespace::NativeNamespace;
 use crate::{
-    types::{ExecutionResult, RunCommandOptions, RunScriptOptions},
+    constants::{NODE_CONFIG_DIR, NODE_DATA_DIR, NODE_RELAY_DATA_DIR, NODE_SCRIPTS_DIR},
+    types::{ExecutionResult, RunCommandOptions, RunScriptOptions, TransferedFile},
     ProviderError, ProviderNode,
 };
 
-#[derive(Clone)]
-pub(super) struct NativeNode<FS, PM>
+pub(super) struct NativeNode<FS>
 where
     FS: FileSystem + Send + Sync + Clone,
-    PM: ProcessManager + Send + Sync + Clone,
 {
-    pub(super) name: String,
-    pub(super) program: String,
-    pub(super) args: Vec<String>,
-    pub(super) env: Vec<(String, String)>,
-    pub(super) base_dir: PathBuf,
-    pub(super) config_dir: PathBuf,
-    pub(super) data_dir: PathBuf,
-    pub(super) relay_data_dir: PathBuf,
-    pub(super) scripts_dir: PathBuf,
-    pub(super) log_path: PathBuf,
-    pub(super) inner: Arc<RwLock<NativeNodeInner>>,
-    pub(super) filesystem: FS,
-    pub(super) process_manager: PM,
-    pub(super) namespace: WeakNativeNamespace<FS, PM>,
+    namespace: Weak<NativeNamespace<FS>>,
```

Comment:
I'm now sure if we need to have a ref to the namespace in the node, in some cases I think we will need the `namespace name` but not the whole info.